crates/re_arrow_store/src/store_sanity.rs

```diff
@@ -8,6 +8,20 @@ use crate::{DataStore, IndexedBucket, IndexedBucketInner, IndexedTable, Persiste
 
 #[derive(thiserror::Error, Debug)]
 pub enum SanityError {
```

Comment:
Please add a doscstring explaining this `enum` (i.e. that the error can only happen due to bugs in `re_arrow_store`)